python/pyspark/sql/pandas/serializers.py

```diff
@@ -1630,8 +1633,23 @@ def row_stream():
                         yield (batch_key, row)
 
             for batch_key, group_rows in groupby(row_stream(), key=lambda x: x[0]):
-                df = pd.DataFrame([row for _, row in group_rows])
-                yield (batch_key, df)
+                rows = []
```

Comment:
@HyukjinKwon @zhengruifeng 
What do you think about the code? We limit the size of Arrow RecordBatch in task thread when sending to Python worker, and @zeruibao added this to re-align the size for Pandas DataFrame. Did we do this in other UDF? Is it beneficial or probably over-thinking?

Reply:
> Did we do this in other UDF?

I don't think so

> Is it beneficial or probably over-thinking?

I remember @HyukjinKwon discussed it before, it should be beneficial if the size is properly estimated